test/Controller/ProfileControllerTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace test\eLife\Journal\Controller;
+
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Response;
+use Symfony\Component\BrowserKit\Client;
+
+final class ProfileControllerTest extends PageTestCase
+{
+    /**
+     * @test
+     */
+    public function it_displays_a_404_if_the_feature_flag_is_disabled()
+    {
+        $client = static::createClient();
+        $client->getCookieJar()->clear();
+
+        $client->request('GET', $this->getUrl());
+
+        $this->assertSame(404, $client->getResponse()->getStatusCode());
+    }
+
+    /**
+     * @test
+     */
+    public function it_displays_a_profile_page()
+    {
+        $client = static::createClient();
+
+        $crawler = $client->request('GET', $this->getUrl());
+
+        $this->assertSame(200, $client->getResponse()->getStatusCode());
+        $this->assertSame('Preferred Name', $crawler->filter('.content-header-profile__display_name')->text());
+        $this->assertEmpty($crawler->filter('.content-header-profile__details'));
+    }
+
+    /**
+     * @test
+     */
+    public function it_has_metadata()
+    {
+        $client = static::createClient();
+
+        $client->request('GET', '/?open-sesame');
+
+        $crawler = $client->request('GET', $this->getUrl().'?foo');
```

Comment:
why the `?foo`?

Reply:
For the `link[rel="canonical"]` assertion.